typing/includemod.ml

```diff
@@ -509,9 +514,9 @@ and check_modtype_equiv ~loc env ~mark cxt mty1 mty2 =
 
 let can_alias env path =
   let rec no_apply = function
-    | Pident _ -> true
-    | Pdot(p, _, _) -> no_apply p
-    | Papply _ -> false
+    | Path.Pident _ -> true
+    | Path.Pdot(p, _, _) -> no_apply p
+    | Path.Papply _ -> false
```

Comment:
FWIW, using disambiguation helps a lot for this kind of thing:
```
let can_alias env path =
  let rec no_apply (path : Path.t) =
    match path with
 ￼   | Pident _ -> true
    | Pdot(p, _, _) -> no_apply p
    | Papply _ -> false
  in
  ...
```

Reply:
Unfortunately, this part of the compiler has warning 40 turned on (and fatal). I think changing default warning in the implementation of the typechecker shouldn't be discussed in this PR.